www/js/survey/enketo-survey-service.js

```diff
@@ -0,0 +1,137 @@
+angular.module('emission.enketo-survey.service', [
+  'ionic',
+  'emission.services',
+  'emission.plugin.logger',
+  'emission.tripconfirm.service',
+])
+.factory('EnketoSurvey', function(
+  $window, $http, UnifiedDataLoader, Logger,
+  ConfirmHelper
+) {
+  var __form = null;
+  var __session = {};
+  var __form_location = null;
+  var __loaded_form = null;
+  var __loaded_model = null;
+
+  function init(formLocParam, optsParam) {
+    __form = null;
+    __session = {};
+    __form_location = formLocParam;
+    __loaded_form = null;
+    __loaded_model = null;
+
+    const opts = JSON.parse(optsParam);
+    if (opts && opts.session) {
+      __session = opts.session;
+    }
+
+    return $http.get(__form_location)
+    .then(function(form_json) {
+      __loaded_form = form_json.data.form;
+      __loaded_model = form_json.data.model;
+    });
+  }
+
+  function _loadForm(opts = {}) {
+    var formSelector = 'form.or:eq(0)';
+    var data = {
+      // required string of the default instance defined in the XForm
+      modelStr: __loaded_model,
+      // optional string of an existing instance to be edited
+      instanceStr: opts.instanceStr || null,
+      submitted: opts.submitted || false,
+      external: opts.external || [],
+      session: opts.session || {}
+    };
+    __form = new $window.FormModule( formSelector, data, {});
+    var loadErrors = __form.init();
+    return loadErrors;
+  }
+
+  function _restoreAnswer(answers) {
+    const answer = ConfirmHelper.getUserInputForTrip(__session.trip_properties, answers);
+    return (!answer) ? null : answer.data.survey_result;
+  }
+
+  function _parseAnswerByTagName(answerXml, tagName) {
+    const vals = answerXml.getElementsByTagName(tagName);
+    const val = vals.length ? vals[0].innerHTML : null;
+    if (!val) return '<null>';
+    return val.replace(/_/g, ' ');
+  }
+
+  function getAllSurveyAnswers(key = 'manual/confirm_survey', opts = {}) {
+    const _opts_populateLabels = opts.populateLabels || false;
+
+    const tq = $window.cordova.plugins.BEMUserCache.getAllTimeQuery();
+    return UnifiedDataLoader.getUnifiedMessagesForInterval(key, tq)
+    .then(function(answers){
+      if (!_opts_populateLabels) return answers;
+
+      const xmlParser = new $window.DOMParser();
+      if (key === 'manual/confirm_survey') {
+        return answers.map(function(answer){
+          const xmlStr = answer.data.survey_result;
+          const xml = xmlParser.parseFromString(xmlStr, 'text/xml');
+          // Data injection
+          answer.travel_mode_main = _parseAnswerByTagName(xml, 'travel_mode_main');
```

Comment:
Potential enhancement: now that you have pulled this out into a separate function, you could theoretically create a JSON file where users can specify the tags and potentially how to format them into the string. Then it is easier for other travel survey people to customize the displayed button without modifying code.

e.g.

```
{'extracted_tags': ['travel_mode_main', 'o_purpose_main', 'd_purpose_main'],
 'display': '<b>{{tripgj.userSurveyAnswer.extracted_tag[0]}}</b> from <b>{{tripgj.userSurveyAnswer.extracted_tag[1]}}</b> to <b>{{tripgj.userSurveyAnswer.extracted_tag[2]}}</b>'
}
```

This is definitely optional, depending on how easy you want to make it for future deployers to customize the UX and how comfortable you think they will be with code v/s editing a JSON file.

And even if you want to do it, I would suggest doing so as part of a separate PR. This one is complex enough already and it works.

The goal is to go from working -> working...